addons/sale_loyalty/models/sale_order.py

```diff
@@ -673,7 +673,7 @@ def _get_claimable_rewards(self, forced_coupons=None):
                     continue
                 # Discounts are not allowed if the total is zero unless there is a payment reward, in which case we allow discounts.
                 # If the total is 0 again without the payment reward it will be removed.
-                if reward.reward_type == 'discount' and total_is_zero and (not has_payment_reward or reward.program_id.is_payment_program):
+                if reward.reward_type == 'discount' and (total_is_zero and (not has_payment_reward or reward.program_id.is_payment_program) or coupon in self.order_line.coupon_id):
```

Comment:
It'll be more readable in 2 separate statements, especially since it's not the same case =]
Also, try to keep the line under 100 chars, and it might also be interesting to add a comment to specify `# already applied` about that coupon